packages/Makefile

```diff
@@ -15,7 +15,12 @@ all: pyodide-build
 	--log-dir=./build-logs
 
 pyodide-build: ../pyodide-build/pyodide_build/**
-	$(HOSTPYTHON) -m pip install --prefix=$(shell $(HOSTPYTHON) -m site --user-base) -e ../pyodide-build
+ifdef VIRTUAL_ENV
+	export PREFIX=""
+else
+	export PREFIX="--prefix=$(shell $(HOSTPYTHON) -m site --user-base)"
+endif
+	$(HOSTPYTHON) -m pip install $(PREFIX) -e ../pyodide-build
```

Comment:
Actually, I wonder if we should just ask users to pip install it before running make and get rid of the prefix altogether (users should use a virtualenv or a  conda env).

Reply:
Yes, I think that is a good idea.

However, even if we ask users to install `pyodide-build` themselves before running `make`, this issue (editable install not working with pyproject.toml) will keep happening unless we stop using editable install.

Reply:
> this issue (editable install not working with pyproject.toml) will keep happening unless we stop using editable install.

I know there was some trouble with it, but I think they disabled build isolation with editable installs now ? In any case, I have no issues with using editable installs with pyproject.toml in other regular Python packages.

Reply:
Hmm, yes I can also build `pyodide-build` in editable install mode without setting `--prefix` now.

I am testing with multiple Python / pip / setuptools versions, and it looks like Python >= 3.7 successes while 3.6 is failing to install in editable mode. I'm not sure which change made it available... I can't find an explicit changelog for this.

So maybe we can just remove `--prefix` and tell people to upgrade `pip` and `setuptools` if it fails? WDYT?

Reply:
Yes, let's do that. The minimal requirements for setuptools [is already in pyproject.toml](https://github.com/pyodide/pyodide/blob/main/pyodide-build/pyproject.toml#L2) but yes, it won't work with old pip versions. It's up to the user to have a reasonably recent pip.